### DISCUSSION TITLE
Lesson 8: brownie test seems to get stuck on test_api_consumer.py

### DISCUSSION THREAD

**Discussion by klovertexarkana** [smartcontractkit/full-blockchain-solidity-course-py#1828] [Q&A]
Running "brownie test", after pulling the chainlink mix from github, creates a testing session that never seems to finish. The terminal gets stuck here:

=============================test session starts======================================
platform linux -- Python 3.8.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/harry/Dropbox/crypto/web_dev_py/demos/mixes/chainlink
plugins: eth-brownie-1.19.1, xdist-1.34.0, hypothesis-6.27.3, web3-5.30.0, forked-1.4.0
collected 7 items                                                                                                                                                            

Launching 'ganache-cli --accounts 10 --hardfork istanbul --gasLimit 12000000 --mnemonic brownie --port 8545'...

tests/test_api_consumer.py

**Comment by cromewar** [comment]
Hello @klovertexarkana 
Please we are going to need more information on this, share the test script and also the command you are using to run the test.

**Comment by klovertexarkana** [comment]
Test script is the same script when you run "brownie bake chainlink-mix" from within your demos/mixes folder:

```Python
import time

import pytest
from brownie import APIConsumer, network, config
from scripts.helpful_scripts import (
    BLOCK_CONFIRMATIONS_FOR_VERIFICATION,
    LOCAL_BLOCKCHAIN_ENVIRONMENTS,
    get_account,
    listen_for_event,
    get_contract,
    fund_with_link,
)


@pytest.fixture
def deploy_api_contract(get_job_id, chainlink_fee):
    # Arrange / Act
    api_consumer = APIConsumer.deploy(
        get_contract("oracle").address,
        get_job_id,
        chainlink_fee,
        get_contract("link_token").address,
        {"from": get_account()},
    )
    api_consumer.tx.wait(BLOCK_CONFIRMATIONS_FOR_VERIFICATION)
    # Assert
    assert api_consumer is not None
    return api_consumer


def test_send_api_request_local(
    deploy_api_contract,
    chainlink_fee,
    get_data,
):
    # Arrange
    if network.show_active() not in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
        pytest.skip("Only for local testing")
    api_contract = deploy_api_contract
    get_contract("link_token").transfer(
        api_contract.address, chainlink_fee * 2, {"from": get_account()}
    )
    # Act
    transaction_receipt = api_contract.requestVolumeData({"from": get_account()})
    requestId = transaction_receipt.events["ChainlinkRequested"]["id"]
    # Assert
    get_contract("oracle").fulfillOracleRequest(
        requestId, get_data, {"from": get_account()}
    )
    assert isinstance(api_contract.volume(), int)
    assert api_contract.volume() > 0


def test_send_api_request_testnet(deploy_api_contract, chainlink_fee):
    # Arrange
    if network.show_active() not in ["kovan", "rinkeby", "goerli", "mainnet"]:
        pytest.skip("Only for testnet testing")
    api_contract = deploy_api_contract

    if config["networks"][network.show_active()].get("verify", False):
        APIConsumer.publish_source(api_contract)

    tx = fund_with_link(api_contract.address, amount=chainlink_fee)
    tx.wait(1)
    # Act
    transaction = api_contract.requestVolumeData({"from": get_account()})
    transaction.wait(1)

    # Assert
    event_response = listen_for_event(api_contract, "DataFullfilled")
    assert event_response.event is not None
    assert isinstance(api_contract.volume(), int)
    assert api_contract.volume() > 0
```

And the command is "brownie test". Once brownie test command is executed my terminal stalls with the output I detailed in the orginal post.

**Reply by cromewar** [reply]
Seems everything ok, did you name the python file using the keyword tests?

**Reply by klovertexarkana** [reply]
> Seems everything ok, did you name the python file using the keyword tests?

I've done it both ways where I just type "brownie test" as well as "brownie test test_api_consumer.py" both give the same results where the test just stalls.

Could the "@pytest.fixtures" line in test_api_consumer.py be an issue? Do I need to install something for this? I'm not familiar w/ the syntax and don't remember this be talked about in prior lessons.

**Reply by cromewar** [reply]
I was reviewing your code and I cannot find anything wrong, could you use the flag `-s` on the test in order for it to go step by step and share here the result?

**Reply by klovertexarkana** [reply]
> I was reviewing your code and I cannot find anything wrong, could you use the flag `-s` on the test in order for it to go step by step and share here the result?

Here is the output:

plugins: eth-brownie-1.19.1, xdist-1.34.0, hypothesis-6.27.3, web3-5.30.0, forked-1.4.0
collected 7 items                                                                                                                                                            

Launching 'ganache-cli --accounts 10 --hardfork istanbul --gasLimit 12000000 --mnemonic brownie --port 8545'...

tests/test_api_consumer.py::test_send_api_request_local RUNNING
The active network is development
Deploying Mocks...
Deploying Mock Link Token...
Deploying Mock Price Feed...
Deployed to 0x602C71e4DAC47a042Ee7f46E0aee17F94A3bA0B6
Deploying Mock VRFCoordinator...
Deployed to 0xE7eD6747FaC5360f88a2EFC03E00d25789F69291
Deploying Mock Oracle...
Deployed to 0x6951b5Bd815043E3F842c1b026b0Fa888Cc2DD85
Deploying Mock Operator...
Deployed to 0xe0aA552A10d7EC8760Fc6c246D391E698a82dDf9
Mocks Deployed!

**Reply by cromewar** [reply]
Seems you have various outputs for printing but its not going trough the asserts. So, could you please share a screenshot of the folder structure on your project.

**Reply by klovertexarkana** [reply]
<img width="349" alt="image" src="https://user-images.githubusercontent.com/97547512/217321625-2cf05027-32b0-4b57-b095-b03b59f6624e.png">

**Reply by cromewar** [reply]
Haven't found any issues so far, have you being able to solve this?

**Reply by klovertexarkana** [reply]
> Haven't found any issues so far, have you being able to solve this?

Nope. Still getting stuck at the same spot. The only thing I've done differently from the video is I'm not using Visual Studio Code. That being said I have never run into any trouble w/ any of the previous testing throughout the entire video.

**Reply by cromewar** [reply]
Just a question if you clone the code repo, do you get the same result.

**Reply by klovertexarkana** [reply]
Yea even if I clone it gets stuck in the same spot. 

I guess I could use VS code and see if that solves the problem. Even if it does, I'd be curious though why this is the first time I've had a problem testing. Like what part of these tests isn't working correctly w/ my current IDE.

**Reply by cromewar** [reply]
I suppose the problem should be on the python installation, could you create a new venv and install just brownie on it? try running the test again, does it works?

**Reply by klovertexarkana** [reply]
No even if I create a new venv and install just brownie it gets stuck on that same test. I guess the problem is just my IDE then?

**Reply by cromewar** [reply]
Could be, as I always use vscode and I tried to reproduce the error without success, could you try with vscode and tell us if the problem persists?

btw: should be a problem as the code will anyways run trough a terminal, but let's try it.
